### PR TITLE
feat: support transaction

### PR DESCRIPTION
**Pull request by hantmac**
- part of https://github.com/datafuselabs/databend/issues/14789

 

- [x] waiting for https://github.com/datafuselabs/databend/pull/14802

### CODE REVIEW

**Comment by hantmac** [comment]
cc @youngsofun plz help review it 😄

File: transaction.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+		tx.dc.batchInsert = nil
+	}()
+	if tx.dc.batchMode && tx.dc.batchInsert != nil {
+		err = tx.dc.batchInsert()
+		if err != nil {
+			return
+		}
+	}
+	// complicity with old server version
+	if tx.dc.rest.sessionState.TxnState != "" {
```
Comment (youngsofun) [inline transaction.go:25]: check null instead of “” （or both）？

Comment (hantmac) [reply transaction.go:25]: make sense.

Comment (youngsofun) [reply transaction.go:25]: rollback is need the same check?

**Comment by youngsofun** [comment]
test 
1.   txn fail (once a stmt fail, the remaining fail untill commit/rollback)
2. interleaved txns: https://github.com/SkyFan2002/test-scripts/blob/main/txn/src/main.rs

**Review by youngsofun — APPROVED** [review]

**Pull request merged**